test/integration/server.cc

```diff
@@ -78,14 +78,7 @@ void IntegrationTestServer::start(const Network::Address::IpVersion version,
 }
 
 IntegrationTestServer::~IntegrationTestServer() {
-  ENVOY_LOG(info, "stopping integration test server");
-
-  if (admin_address_ != nullptr) {
-    BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
-        admin_address_, "POST", "/quitquitquit", "", Http::CodecClient::Type::HTTP1);
-    EXPECT_TRUE(response->complete());
-    EXPECT_STREQ("200", response->headers().Status()->value().c_str());
-  }
+  // Derived class must have shutdown server.
```

Comment:
Does it make sense to assert this?

Reply:
At this point in the code, the derived class has been torn down.  That specifically means that calls to server() will be resolved in the vtable of the base class (and yeah, I dunno what that means if it's marked as PURE in the base class).  So the only way I see to do this would be for the derived class to call a function on the base class to leave a breadcrumb saying "No more server", which doesn't seem worth it to me.  Do you see another way to assert the derived class has shutdown the server?  (Or think I should use the breadcrumb option?)

Reply:
I think it's OK. Can you put a comment on the ~IntegrationTestServer() in the header file indicating that the derived class is responsible for shutting down the server?

Reply:
Done.  Note that there's also such a note in the comment for createAndRunEnvoyServer; I'm leaving that one in place since I think (hope) that implementors of the derived class will read the comments on the pure virtual methods since they have no choice but to implement those methods :-}.